x-pack/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {
+  Position,
+  Chart,
+  Settings,
+  Axis,
+  LineSeries,
+  getAxisId,
+  getSpecId,
+  AreaSeries,
+  BarSeries,
+} from '@elastic/charts';
+import { Datatable } from '../../../canvas/canvas_plugin_src/functions/types';
+
+/**
+ * This file contains TypeScript type definitions and their equivalent expression
+ * definitions, for configuring and rendering an XY chart. The XY chart serves
+ * triple duty as a bar, line, or area chart.
+ *
+ * The xy_chart expression function serves mostly as a passthrough to the xy_chart_renderer
+ * which does the heavy-lifting.
+ */
+
+/**
+ * Configuration for the chart's legend.
+ */
+export interface LegendConfig {
+  isVisible: boolean;
+  position: Position;
+}
+
+/**
+ * LegendConfig as an expression type.
+ */
+export const legendConfig = {
+  name: 'legendConfig',
+  aliases: [],
+  type: 'legendConfig',
+  help: `Configure a chart's legend`,
+  context: {
+    types: ['null'],
+  },
+  args: {
+    isVisible: { types: ['boolean'] },
+    position: { types: ['string'] },
+  },
+  fn: function fn(_context: any, args: LegendConfig) {
+    return {
+      type: 'legendConfig',
+      ...args,
+    };
+  },
+};
+
+/**
+ * Properties common to all axes.
+ */
+interface AxisConfig {
+  title: string;
+  showGridlines: boolean;
+  position: Position;
+}
+
+/**
+ * AxisConfig expression definition.
+ */
+const axisConfig = {
+  title: { types: ['string'] },
+  showGridlines: { types: ['boolean'] },
+  position: { types: ['string'] },
+};
```

Comment:
Given the existing AxisConfig, I'm worried about confusing naming. I assume each visualization will be defining its own expressions and types, so it might be worth it to distinguish in the naming between each visualization and potentially also with a lens prefix.

Reply:
I don't generally like using a prefix. That's what modules are for. It's OK for there to be similar or even the same name for similar / same concepts across Kibana, as long as they are contained in their own modules. That said, I do like the idea of disambiguating things. Can you think of a good way to do that without prefixes?